src/hooks/mutations/useMintToken.ts

```diff
@@ -0,0 +1,26 @@
+import { useWalletInterface } from "@/services/wallets/useWalletInterface";
+import { useMutation } from "@tanstack/react-query";
+import { writeTokenMint } from "@/services/contracts/wagmiGenActions";
+import { AccountId } from "@hashgraph/sdk";
+import { WalletInterface } from "@/services/wallets/walletInterface";
+import { MintTokenRequest } from "@/types/types";
+
+export function useMintToken() {
+  const { walletInterface } = useWalletInterface();
+
+  return useMutation({
+    mutationFn: async ({ token, address, value }: MintTokenRequest) => {
+      const evmAddress = await walletInterface?.getEvmAccountAddress(
+        AccountId.fromString(address as string),
+      );
+      if (!evmAddress) return null;
```

Comment:
Maybe we should throw here? Whats the best way for the inside of murations

Reply:
I'm planning to refactor this part a bit in one of my todos - so evm address will be a part of wallet interface, so in this case mutation will have correct evm address or will not be called.